src/middlewared/middlewared/plugins/cron.py

```diff
@@ -0,0 +1,175 @@
+from middlewared.validators import Range
+
+
+class CronJobService(CRUDService):
+
+    class Config:
+        datastore = 'tasks.cronjob'
+        datastore_extend = 'cron.job.cron_extend'
+        datastore_prefix = 'cron_'
+        namespace = 'cron.job'
+
+    async def cron_extend(self, data):
+        if data['dayweek'] == '*':
+            data['dayweek'] = [i for i in range(1, 8)]
+        else:
+            data['dayweek'] = [int(v) for v in data['dayweek'].split(',')]
+        if data['month'] == '*':
+            data['month'] = [i for i in range(1, 13)]
+        else:
+            data['month'] = [int(v) for v in data['month'].split(',')]
+        return data
+
+    async def validate_data(self, data, schema):
+        verrors = ValidationErrors()
+
+        command = data.get('command')
+        if not command:
+            verrors.add(
+                f'{schema}.command',
+                'This field is required'
+            )
+
+        weekdays = data.get('dayweek')
+        if weekdays:
+            if len([day for day in weekdays if day not in range(1, 8)]) > 0:
+                verrors.add(
+                    f'{schema}.weekday',
+                    'The week days should be in range of 1-7 inclusive'
+                )
+            elif len(weekdays) == 7:
+                data['dayweek'] = '*'
+            else:
+                data['dayweek'] = ','.join([str(day) for day in weekdays])
+        else:
+            data['dayweek'] = ''
+
+        months = data.get('month')
+        if months:
+            if len([month for month in months if month not in range(1, 13)]) > 0:
+                verrors.add(
+                    f'{schema}.month',
+                    'The months should be in range of 1-12 inclusive'
+                )
+            elif len(months) == 12:
+                data['month'] = '*'
+            else:
+                data['month'] = ','.join([str(month) for month in months])
+        else:
+            data['month'] = ''
+
+        user = data.get('user')
+        if not user:
+            verrors.add(
+                f'{schema}.user',
+                'This field is required'
+            )
+        else:
+            if not (
+                await self.middleware.call(
+                    'notifier.get_user_object',
+                    user
+                )
+            ):
+                verrors.add(
+                    f'{schema}.user',
+                    'Specified user does not exist'
+                )
+        return verrors, data
+
+    @accepts(
+        Dict(
+            'cron_job_create',
+            Bool('enabled'),
+            Bool('stderr'),
+            Bool('stdout'),
+            List('dayweek', items=[Int('dayweek')]),
+            List('month', items=[Int('month')]),
+            Str('command'),
+            Str('daymonth'),
+            Str('description'),
+            Str('hour'),
+            Str('minute'),
+            Str('user'),
```

Comment:
dayweek and month seem to be a bit inconsistent with the rest of the option for the API.

I think dayweek and month should just be Str, then every option could just follow the crontab syntax, e.g. `*/4`, `5,7,8,9`, `*`. Thoughts?